lib/Server.js

```diff
@@ -319,7 +319,11 @@ class Server {
         } else {
           next();
         }
-      });
+      };
+
+      this.app.use(handle);
+      // Also forward error requests to the proxy so it can handle them.
+      this.app.use((error, req, res, next) => handle(req, res, next));
```

Comment:
It is create two instances of proxy that was reduce performance

Reply:
I don't see how that will hurt performance.

The proxy middleware is created once, outside of `handle`:

```ts
      proxyMiddleware = getProxyMiddleware(proxyConfig);
```

… and the result is used in `handle` on each request, be it a non-error or error request.

Reply:
Each request now run `handle` (https://github.com/webpack/webpack-dev-server/blob/29a822ea8426820617745b3f355c344e84f7c01b/lib/Server.js#L291)  twice so we potentially send require to middleware twice https://github.com/webpack/webpack-dev-server/blob/29a822ea8426820617745b3f355c344e84f7c01b/lib/Server.js#L318

Reply:
I don't believe that's true. `handle` will only run once:

- If the request has not errored, `handle` will be called via `app.use(handle)`.
- If the request has errored, `handle` will be called via `app.use((error, req, res, next) => handle(req, res, next))`.

You can see this for yourself if you add some logging to `handle`.

Reply:
Hm, looks you are right https://expressjs.com/en/guide/error-handling.html